File: test-app.js
import { chromium } from "playwright";

(async () => {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext();
  const page = await context.newPage();

  const consoleMessages = [];
  const errors = [];

  page.on("console", (msg) => {
    consoleMessages.push({ type: msg.type(), text: msg.text(), location: msg.location() });
    if (msg.type() === "error") {
      errors.push({ message: msg.text(), location: msg.location() });
    }
  });

  page.on("pageerror", (error) => {
    errors.push({ message: error.message, stack: error.stack });
  });

  try {
    console.log("Navigating to http://localhost:3000...");
    await page.goto("http://localhost:3000", { waitUntil: "networkidle", timeout: 10000 });

    console.log("Page loaded successfully!");
    console.log("Page title:", await page.title());

    // Wait a bit for any async operations
    await page.waitForTimeout(3000);

    console.log("\n=== Console Errors with Location ===");
    errors.forEach((err) => {
      console.log("ERROR:", err.message);
      if (err.location) {
        console.log("Location:", JSON.stringify(err.location));
      }
      if (err.stack) {
        console.log("Stack:", err.stack);
      }
    });

    if (errors.length === 0) {
      console.log("No errors detected!");
    }
  } catch (e) {
    console.error("Test failed:", e.message);
  } finally {
    await browser.close();
  }
})();

File: test-app.py
#!/usr/bin/env python3
"""
Test the Llama Proxy Dashboard application with Playwright
"""

from playwright.sync_api import sync_playwright
import sys


def test_application():
    errors = []

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        # Capture console errors
        def handle_console(msg):
            if msg.type == "error":
                errors.append(f"Console Error: {msg.text}")

        page.on("console", handle_console)

        # Navigate to the app
        print("Navigating to http://localhost:3000...")
        page.goto("http://localhost:3000")

        # Wait for the page to fully load
        page.wait_for_load_state("networkidle")
        print("Page loaded, waiting for app to initialize...")

        # Wait a bit for React/JS to fully initialize
        page.wait_for_timeout(3000)

        # Check if main elements exist
        print("Checking for main application elements...")

        # Check for app container
        app = page.locator("#app")
        if app.count() > 0:
            print("✅ App container found")
        else:
            errors.append("❌ App container #app not found")

        # Check for dashboard
        dashboard = page.locator(".dashboard-page, #dashboard")
        if dashboard.count() > 0:
            print("✅ Dashboard found")
        else:
            print("⚠️ Dashboard not found (may be on different page)")

        # Check for navigation
        nav = page.locator("nav, .nav, [role='navigation']")
        if nav.count() > 0:
            print("✅ Navigation found")
        else:
            print("⚠️ Navigation not found")

        # Take a screenshot for inspection
        page.screenshot(path="/tmp/app-screenshot.png", full_page=True)
        print("📸 Screenshot saved to /tmp/app-screenshot.png")

        # Print page content summary
        content = page.content()
        if "Dashboard" in content or "dashboard" in content.lower():
            print("✅ Dashboard text found in page")
        else:
            print("⚠️ Dashboard text not found in page")

        browser.close()

    # Report results
    print("\n" + "=" * 50)
    print("TEST RESULTS")
    print("=" * 50)

    if errors:
        print(f"❌ {len(errors)} error(s) found:")
        for error in errors:
            print(f"  - {error}")
        return 1
    else:
        print("✅ No console errors detected!")
        print("✅ Application loaded successfully!")
        return 0


if __name__ == "__main__":
    sys.exit(test_application())
